Expensify/App
# Set the focus on Edit Box after edit action.

### Details
1. Remove modal visibility status from the Context Menu. which is needed to make the edited comment as a prime actor for focus.
2. Delayed the edit action after the Context menu is completely hidden which let us handle the focus. It removes the need for the SetTimeout hack.
3. Set the focus on the Edit box after each edit action.
4. Move the focus back to the main Composer when the Edit box is closed.

### Fixed Issues

Fixes #3017
fixes #3309 

### Tests | QA Steps (Web|Desktop)
* Sign in to the e.chat staging web app
* Right-click the text and choose Edit Comment
* Textbox should automatically get focused.
* Click Edit comment from Hover menu. Edit box should automatically get focused.
* Send a message, then hit the `ArrowUp` key. Edit box of your latest message should automatically get focused.
* Hit the `Esc` key, and the main `ReportActionCompose` should regain focus.

### Tests | QA Steps(mWeb|Android|IOS)
* Sign in to the e.chat staging web app
* Hold press any text message sent by you and choose Edit Comment
* Textbox should automatically get focused.

### Tests | QA Steps (Web|Desktop) for #3309
* Sign in to the e.chat staging web app
* Right-click the text and choose Edit Comment
* Textbox should automatically get focused.
* Click Edit comment from Hover menu. Edit box should automatically get focused.
* Save changes , main chat composer should regain focus.
* cancel Edit, main chat composer should regain focus.
* Click edit comment for any message.
* Hit the `Esc` key, and the main chat composer should regain focus.

### Tested On


### Screenshots


#### Web



https://user-images.githubusercontent.com/24370807/121236427-93d76480-c8b3-11eb-9f20-9b8787a71e93.mp4


#### Mobile Web


https://user-images.githubusercontent.com/24370807/121238132-64c1f280-c8b5-11eb-868a-bbb082f05584.mp4


#### Desktop



https://user-images.githubusercontent.com/24370807/121236814-fa5c8280-c8b3-11eb-9c5b-4afd93dfcc4d.mp4

#### iOS


#### Android


https://user-images.githubusercontent.com/24370807/121239234-9ab3a680-c8b6-11eb-9209-a5ee0eaaade9.mp4
